src/helpers.c

```diff
@@ -595,6 +595,32 @@ int swupd_init(void)
 	return ret;
 }
 
+/* The strtol function is commonly used to convert a string to a number and
+ * the result is frequently stored in an int type. Type casting a long to an
+ * int can cause overflows. This function sets 'err' to a nonzero value when
+ * either the strtol return value or the long to int type conversion overflows.
+*/
+int strtoi_err(const char *str, char **endptr, int base, int *err)
+{
+	long num;
+
+	errno = 0;
+	num = strtol(str, endptr, base);
+	*err = errno;
+
+	/* When the return value of strtol overflows the int type, don't overflow
+	 * and return an error. */
+	if (num > INT_MAX) {
+		num = INT_MAX;
+		*err = -1;
```

Comment:
You can use the same codes of errno table here. -EOVERFLOW is more appropriate than -1 (EPERM)

Reply:
And say in the doc above that you are running negative numbers correspondent to errno table.